src/test/ui/consts/const-num-variants.rs

```diff
@@ -0,0 +1,28 @@
+// run-pass
+#![allow(dead_code)]
+#![feature(const_num_variants)]
+#![feature(num_variants)]
+
+use std::mem::num_variants;
+
+enum Foo {
+    A,
+    B,
+    C,
+}
+
+enum Bar {
+    A,
+    B,
+    C,
+    D(usize),
+    E { field_1: usize, field_2: Foo },
+}
+
+const TEST_FOO: usize = num_variants::<Foo>();
+const TEST_BAR: usize = num_variants::<Bar>();
+
+fn main() {
+    assert_eq!(TEST_FOO, 3);
+    assert_eq!(TEST_BAR, 5);
```

Comment:
```suggestion
    assert_eq!(TEST_BAR, 5);
    assert_eq!(num_variants::<Bar>(), 5);
    assert_eq!(num_variants::<Option>(), 2);
```
Tests this function outside of a const context and for extern crates

Reply:
should be something like `Option<char>` rather than just `Option`.